consensus/src/experimental/buffer_manager.rs

```diff
@@ -172,38 +174,19 @@ impl StateManager {
     /// check if the items at and after the execution root is already executed
     /// if yes, move the execution root to the first *unexecuted* item.
     /// if there is no such item, set it to none.
-    fn try_advance_executed_root(&mut self) {
-        let mut cursor = self.execution_root.clone();
-        while cursor.is_some() {
-            {
-                let buffer_item = get_elem(&cursor);
-                if !matches!(&*buffer_item, BufferItem::Executed(_)) {
-                    break;
-                }
-            }
-            cursor = get_next(&cursor);
-        }
-        self.execution_root = cursor;
+    fn advance_executed_root(&mut self) {
+        let cursor = self.execution_root.clone();
+        self.execution_root = find_elem(cursor, |item| !matches!(*item, BufferItem::Executed(_)));
```

Comment:
We probably also need to move the root if the current buffer item is signed. Recall that we allow scattered signed buffer items appear after the signing root:

[Signed] -- [Signed] -- [Executed: Signing Root] -- [Executed] -- [Signed] -- [Executed] -- [Ordered] ...

Now the executed root will stop at the 5-th but it should be at 7-th.